dbms/src/Columns/ColumnString.cpp

```diff
@@ -522,76 +551,84 @@ void ColumnString::countSerializeByteSizeImpl(
                 sizeAt(i));
     }
 
-    if constexpr (has_collator)
+    const size_t size = byte_size.size();
+    size_t max_bytes_one_char = 0;
+    if constexpr (need_decode_collator)
     {
         RUNTIME_CHECK(collator);
+        max_bytes_one_char = collator->sortKeyReservedSpaceMultipler();
+    }
 
-        const size_t size = byte_size.size();
-        const size_t max_bytes_one_char = collator->maxBytesForOneChar();
-        for (size_t i = 0; i < size; ++i)
+    for (size_t i = 0; i < size; ++i)
+    {
+        if constexpr (has_nullmap)
         {
-            assert(sizeAt(i) > 0);
-            if constexpr (count_code_points)
-            {
-                const auto num_char = UTF8::countCodePoints(&chars[offsetAt(i)], sizeAt(i) - 1);
-                // Add 1 for terminating zero.
-                byte_size[i] += sizeof(UInt32) + num_char * max_bytes_one_char + 1;
-            }
-            else
+            if (DB::isNullAt(*nullmap, i))
             {
-                byte_size[i] += sizeof(UInt32) + sizeAt(i);
+                byte_size[i] += sizeof(UInt32) + 1;
```

Comment:
What does `sizeof(UInt32)` and `1` represent respectively?

Reply:
sizeof(UInt32): length of str
1: the default value for ColumnString(a.k.a. '\0')